src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menAverageAge = people
+    .filter(person =>
+      century
+        ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+        : person.sex === 'm')
+    .map(person => person.died - person.born);
+
+  return menAverageAge.reduce((acc, year) => acc + year) / menAverageAge.length;
```

Comment:
let's create a callback function for this logic, you repeat it three times

Reply:
thanks! fixed, but Im not sure if I named it correctly("average")